packages/testing/src/consensus_testing/test_fixtures/fork_choice.py

```diff
@@ -235,6 +239,23 @@ def make_fixture(self) -> ForkChoiceTest:
                     block = signed_block.message.block
                     step._filled_block = block
 
+                    # Validate that the same validator doesn't propose competing blocks
+                    # at the same slot
+                    if block.slot not in slot_proposers:
+                        slot_proposers[block.slot] = set()
+
+                    if block.proposer_index in slot_proposers[block.slot]:
+                        raise ValueError(
+                            f"Step {i}: Validator {block.proposer_index} is proposing multiple "
```

Comment:
We could make this description more concise, something like "Step {i}: Validator {block.proposer_index} is equivocating at slot {block.slot}.\n"